From 0d09ed08a84ec77e01bede24847f4ced4e99057c Mon Sep 17 00:00:00 2001
From: Christopher Larson <chris_larson@mentor.com>
Date: Fri, 29 Sep 2023 17:07:30 -0500
Subject: [PATCH] Ensure that the consumerd configure arguments are used

Currently, the --with-consumerd32-bin, --with-consumerd32-libdir, etc arguments
are not being used by lttng-sessiond, which causes problems with providing a
well configured multilib tracing environment. Ensure that they are used as the
default values in sessiond.

Signed-off-by: Christopher Larson <chris_larson@mentor.com>
---
Upstream-Status: Pending

diff --git a/src/bin/lttng-sessiond/sessiond-config.c b/src/bin/lttng-sessiond/sessiond-config.c
index 05e0ef1..d98876d 100644
--- a/src/bin/lttng-sessiond/sessiond-config.c
+++ b/src/bin/lttng-sessiond/sessiond-config.c
@@ -51,14 +51,14 @@ struct sessiond_config sessiond_config_build_defaults = {
 	.load_session_path.value =		NULL,
 
 	.consumerd32_path.value =		NULL,
-	.consumerd32_bin_path.value =		NULL,
-	.consumerd32_lib_dir.value =		NULL,
+	.consumerd32_bin_path.value =		DEFAULT_USTCONSUMERD32_BIN,
+	.consumerd32_lib_dir.value =		DEFAULT_USTCONSUMERD32_LIBDIR,
 	.consumerd32_err_unix_sock_path.value = NULL,
 	.consumerd32_cmd_unix_sock_path.value = NULL,
 
 	.consumerd64_path.value =		NULL,
-	.consumerd64_bin_path.value =		NULL,
-	.consumerd64_lib_dir.value =		NULL,
+	.consumerd64_bin_path.value =		DEFAULT_USTCONSUMERD64_BIN,
+	.consumerd64_lib_dir.value =		DEFAULT_USTCONSUMERD64_LIBDIR,
 	.consumerd64_err_unix_sock_path.value = NULL,
 	.consumerd64_cmd_unix_sock_path.value = NULL,
 
diff --git a/src/common/defaults.h b/src/common/defaults.h
index 3017a38..9880d13 100644
--- a/src/common/defaults.h
+++ b/src/common/defaults.h
@@ -70,11 +70,15 @@
 #define DEFAULT_KCONSUMERD_ERR_SOCK_PATH        DEFAULT_KCONSUMERD_PATH "/error"
 
 /* UST 64-bit consumer path */
+#define DEFAULT_USTCONSUMERD64_BIN              CONFIG_CONSUMERD64_BIN
+#define DEFAULT_USTCONSUMERD64_LIBDIR           CONFIG_CONSUMERD64_LIBDIR
 #define DEFAULT_USTCONSUMERD64_PATH             DEFAULT_CONSUMERD_RUNDIR "/ustconsumerd64"
 #define DEFAULT_USTCONSUMERD64_CMD_SOCK_PATH    DEFAULT_USTCONSUMERD64_PATH "/command"
 #define DEFAULT_USTCONSUMERD64_ERR_SOCK_PATH    DEFAULT_USTCONSUMERD64_PATH "/error"
 
 /* UST 32-bit consumer path */
+#define DEFAULT_USTCONSUMERD32_BIN              CONFIG_CONSUMERD32_BIN
+#define DEFAULT_USTCONSUMERD32_LIBDIR           CONFIG_CONSUMERD32_LIBDIR
 #define DEFAULT_USTCONSUMERD32_PATH             DEFAULT_CONSUMERD_RUNDIR "/ustconsumerd32"
 #define DEFAULT_USTCONSUMERD32_CMD_SOCK_PATH    DEFAULT_USTCONSUMERD32_PATH "/command"
 #define DEFAULT_USTCONSUMERD32_ERR_SOCK_PATH    DEFAULT_USTCONSUMERD32_PATH "/error"
